marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/Utxo/UtxoIndex.hs

```diff
@@ -97,72 +84,84 @@ tests = testGroup "Spec.Marconi.ChainIndex.Indexers.Utxo"
           propJsonRoundtripUtxoRow
     ]
 
-eventsToRowsRoundTripTest :: Property
-eventsToRowsRoundTripTest  = property $ do
+-- | Round trip UtxoEvents to UtxoRow conversion
+-- The purpose of this test is to show that there is a isomorphism between `UtxoRow` and UtxoEvent.
+propRoundTripEventsToRowConversion :: Property
+propRoundTripEventsToRowConversion  = property $ do
   events <- forAll UtxoGen.genUtxoEvents
-  let f :: C.ChainPoint -> IO (Set C.TxIn)
-      f C.ChainPointAtGenesis = pure  Set.empty
-      f _                     = pure . Utxo.ueInputs $ head events
-      rows = concatMap Utxo.eventsToRows events
+  let
+    txInsMap :: Map C.SlotNo (Set C.TxIn)
+    txInsMap
+      = Data.Map.fromList
+      . concatMap g
+      $ events
+    f :: C.SlotNo -> IO (Set C.TxIn)
+    f sn  = pure $ fromMaybe Set.empty (Data.Map.lookup sn txInsMap)
+    g :: StorableEvent Utxo.UtxoHandle -> [(C.SlotNo, Set C.TxIn)]
+    g (Utxo.UtxoEvent _ ins (C.ChainPoint sn _)) = [ (sn, ins)]
+    g _                                          = []
+
+    postGenesisEvents = filter (\e -> C.ChainPointAtGenesis /= Utxo.ueChainPoint e) events
+    rows :: [Utxo.UtxoRow]
+    rows = concatMap Utxo.eventToRows postGenesisEvents
   computedEvent <- liftIO . Utxo.rowsToEvents f $ rows
-  let postGenesisEvents = filter (\e -> C.ChainPointAtGenesis /= Utxo.ueChainPoint e) events
-  length computedEvent === (length . fmap Utxo.ueChainPoint $ postGenesisEvents)
-  Hedgehog.assert (equivalentLists computedEvent postGenesisEvents)
+  List.sort computedEvent === List.sort postGenesisEvents
+  List.sort computedEvent === List.sort events
 
 -- Insert Utxo events in storage, and retreive the events
---
-utxoStorageTest :: Property
-utxoStorageTest = property $ do
+-- Note:
+-- we are intetested in testing the in-memory storage only for this test.
+-- The semantics of disk storage in regards to `ChainPoint.slotNo` are such that:
+--    - database rerieved Spent TxIns is a function of `depth` and `slotNo`, and thus potentially
+--  A consequence of the above is that the set of databaset fetched `Sptnt TxIns` may not match those of the original event
+propSaveToAndRetrieveFromUtxoStore :: Property
+propSaveToAndRetrieveFromUtxoStore = property $ do
+  let depth = Utxo.Depth 2160 -- we set a high valued for depth, to make sure we only test in-memory buffer
   events <- forAll UtxoGen.genUtxoEvents
   (storedEvents :: [StorableEvent Utxo.UtxoHandle]) <-
-    (liftIO . Utxo.open ":memory:") (Utxo.Depth 10)
+    (liftIO $ Utxo.open ":memory:" depth False) -- don't vacuum sqlite
      >>= liftIO . Storable.insertMany events
      >>= liftIO . Storable.getEvents
-  Hedgehog.assert (equivalentLists storedEvents events)
-
--- Insert Utxo events in storage, and retrieve the events by address
```

Comment:
@koslambrou  this test always failed too and was commented out .  
it is replaced with `propUtxoQueryByAddressAndQueryInterval`

Reply:
Actually, my question is a bit different. Most of the existing tests were failing because of the implementation of `equivalentLists`. I know you fixed that in this PR and then correctly changed the property tests. My question is more: if you revert back the fix on the implementation of the querying, is there any of the existing property tests that detect that failure. Basically, have you TDD'd this. I just want to make sure you have a regression test.

Reply:
Yes, I belive so. Here is what I did:
This is the test failure I started with:
```  ┏━━ test/Spec/Marconi/ChainIndex/Indexers/Utxo/UtxoIndex.hs ━━━
            126 ┃ utxoInsertAndQueryTest :: Property
            127 ┃ utxoInsertAndQueryTest = property $ do
            128 ┃   events <- forAll genUtxoEvents
                ┃   │ [ UtxoEvent
                ┃   │     { ueUtxos =
                ┃   │         fromList
                ┃   │           [ Utxo
                ┃   │               { _address =
                ┃   │                   AddressByron
                ┃   │                     (ByronAddress
                ┃   │                        Address
                ┃   │                          { addrRoot =
                ┃   │                              c75ac0f6c2a478087d68308dbbc5f533395f68dbb6d581ffb8cb562b
                ┃   │                          , addrAttributes =
                ┃   │                              Attributes
                ┃   │                                { data_ =
                ┃   │                                    AddrAttributes
                ┃   │                                      { aaVKDerivationPath = Nothing
                ┃   │                                      , aaNetworkMagic = NetworkMainOrStage
                ┃   │                                      }
                ┃   │                                }
                ┃   │                          , addrType = ATVerKey
                ┃   │                          })
```
The failure was from `bad` event -> utxoRow conversion.   

fixed that, and replaced the equivalent test with `===`, and all tests passed.

fixing that function and replacing the equivalent tests,  fixed the remaining tests

Reply:
@kayvank Tried reverting your fix to the old implementation, while keeping all your property tests. None fail with the old implementation, therefore doesn't seem like you added a regression test. Can you validate?